internal/neofs/neofs.go

```diff
@@ -136,6 +136,18 @@ func (x *NeoFS) CreateContainer(ctx context.Context, prm neofs.PrmContainerCreat
 		cnrOptions = append(cnrOptions, container.WithAttribute(attr[0], attr[1]))
 	}
 
+	// https://github.com/nspcc-dev/neofs-s3-gw/issues/435
+	// environment without hh disabling feature will ignore this attribute
+	// environment with hh disabling feature will set proper value from network config
+	hhDisabledValue := "false"
+	if hhDisabled, err := isHomomorphicHashDisabled(ctx, x.pool); err != nil {
+		return nil, err
+	} else if hhDisabled {
+		hhDisabledValue = "true"
+	}
+	cnrOptions = append(cnrOptions, container.WithAttribute(
```

Comment:
Correct, but notice that if there is no requirement to add an attribute on S3-side, we can add it only `if hhDisabled` since missing attribute is equivalent to false.

Reply:
That's nice!